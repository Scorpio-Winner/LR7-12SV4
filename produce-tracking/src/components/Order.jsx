import React from "react";
import { useNavigate } from "react-router-dom";

import { Typography, Button, Box } from '@mui/material';

import classes from "../styles/Order.module.css";

const Order = ({ orderData, orderStatuses, detailsHandler, deleteHandler, readonly }) => {
    const router = useNavigate();

    return (
        <div className={classes.order}>
            <div>
              <Box sx={{ display: 'flex', gap: '7vw' }}>
                <Typography variant="body1" sx={{ fontWeight: 'bold', width: '20px' }}>
                  {orderData.id}
                </Typography>
                <Typography variant="body1" sx={{ width: '170px' }}>
                  {orderData.title}
                </Typography>
                <Typography variant="body1" sx={{ width: '150px' }}>
                  {orderData.status}
                </Typography>
              </Box>
            </div>
            <div>
                <Button
                    variant="outlined"
                    onClick={() => detailsHandler(orderData)}
                    sx={{ marginRight: '8px' }}
                >
                    Details
                </Button>
                {!readonly && <>
                <Button
                    variant="outlined"
                    onClick={() => router(`/orders/edit/${orderData.id}`)}
                    sx={{ marginRight: '8px' }}
                >
                    Edit
                </Button>
                <Button
                    variant="outlined"
                    onClick={() => deleteHandler(orderData.id)}
                >
                    Delete
                </Button>
                </>}
            </div>
        </div>
    );
};

export default Order;