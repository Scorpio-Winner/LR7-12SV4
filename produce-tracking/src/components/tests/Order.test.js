import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Order from '../Order';
import renderer from 'react-test-renderer';

describe('Order', () => {
  const mockOrderData = {
    id: '1',
    title: 'Test Order',
    status: 'Pending',
  };

  const mockDetailsHandler = jest.fn();
  const mockDeleteHandler = jest.fn();

  it('renders correctly', () => {
    const { getByText } = render(
      <Router>
        <Order
          orderData={mockOrderData}
          detailsHandler={mockDetailsHandler}
          deleteHandler={mockDeleteHandler}
          readonly={false}
        />
      </Router>
    );

    expect(getByText('Test Order')).toBeTruthy();
    expect(getByText('Pending')).toBeTruthy();
  });

  it('calls detailsHandler when Details button is clicked', () => {
    const { getByText } = render(
      <Router>
        <Order
          orderData={mockOrderData}
          detailsHandler={mockDetailsHandler}
          deleteHandler={mockDeleteHandler}
          readonly={false}
        />
      </Router>
    );
  
    fireEvent.click(getByText('Details'));
    expect(mockDetailsHandler).toHaveBeenCalledWith(mockOrderData);
  });


  beforeEach(() => {
    localStorage.setItem('role', 'admin');
  });
  
  it('calls deleteHandler when Delete button is clicked', () => {
    const { getByText } = render(
      <Router>
        <Order
          orderData={mockOrderData}
          detailsHandler={mockDetailsHandler}
          deleteHandler={mockDeleteHandler}
          readonly={false}
        />
      </Router>
    );
  
    fireEvent.click(getByText('Delete'));
    expect(mockDeleteHandler).toHaveBeenCalledWith(mockOrderData.id);
  });


  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Order
            orderData={mockOrderData}
            detailsHandler={mockDetailsHandler}
            deleteHandler={mockDeleteHandler}
            readonly={false}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  
});
