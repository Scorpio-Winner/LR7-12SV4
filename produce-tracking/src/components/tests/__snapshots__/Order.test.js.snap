// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order matches the snapshot 1`] = `
<div
  className="order"
>
  <div>
    <div
      className="MuiBox-root css-q9c22"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-dtp3js-MuiTypography-root"
      >
        1
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-19li7pl-MuiTypography-root"
      >
        Test Order
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ytge06-MuiTypography-root"
      >
        Pending
      </p>
    </div>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1wgcmb5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Details
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1wgcmb5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Edit
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
